src/manager.rs

```diff
@@ -0,0 +1,212 @@
+        match (enabled, is_running) {
+            (false, true) => {
+                if let Some(mut runner) = self.runners.remove(&id) {
+                    let (tx, rx) = channel();
+                    runner.stop(move |result| {
+                        tx.send(result).unwrap();
+                    });
+                    if let Err(_) = rx.recv() {
+                        return Err(Error::RunError(RunError::StartStopError(StartStopError::ThreadError)));
+                    }
+                }
+            },
+            (true, false) => {
+                try!(self.start_script(id, &source));
+                let connection = try!(rusqlite::Connection::open(&self.path));
+                try!(connection.execute("UPDATE scripts SET is_enabled = $1 WHERE id = $2",
+                                        &[&enabled, &id.to_string()]));
+            },
+            _ => {
+                // Nothing to do.
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Remove a script entirely, stopping it if necessary.
+    pub fn remove(&mut self, id: &Id<ScriptId>) -> Result<(), Error> {
+        try!(self.set_enabled(id, false));
+        let connection = try!(rusqlite::Connection::open(&self.path));
+        connection.execute("DELETE FROM scripts WHERE id = $1", &[&id.to_string()])
+            .map(|_| ())
+            .map_err(From::from)
+    }
+
+    /// Remove all scripts, stopping any running scripts.
+    /// (If any scripts fail to stop, we store and return those errors in a Vec<Error> so that
+    /// we ensure that the database always gets wiped.)
+    pub fn remove_all(&mut self) -> Result<Vec<Error>, Error> {
+        // Get a copy of the keys (so that we don't borow `self` twice).
+        let keys: Vec<Id<ScriptId>> = self.runners.keys().map(|k| k.clone()).collect();
+        let mut errors = Vec::new();
+        // Remove any running scripts, storing errors for later return so that we
+        // for sure end up nuking the database at the end.
+        for id in keys {
+            if let Err(err) = self.remove(&id) {
+                errors.push(err);
+            }
+        }
+        // Nuke the scripts database.
+        let connection = try!(rusqlite::Connection::open(&self.path));
+        try!(connection.execute("DELETE FROM scripts", &[])
+                .map(|_| ()));
+        Ok(errors)
+    }
+
+    /// Get the number of currently-running scripts.
+    pub fn get_running_count(&self) -> usize {
+        self.runners.len()
+    }
+
+    /// Get the source for a script with the given id.
+    fn get_source(&self, id: &Id<ScriptId>) -> Result<String, Error> {
+        let connection = try!(rusqlite::Connection::open(&self.path));
+        let mut stmt = try!(connection.prepare("SELECT source FROM scripts WHERE id = $1"));
+        let mut rows = try!(stmt.query(&[&id.to_string()]));
+        let first_row = try!(try!(rows.nth(0).ok_or(Error::NoSuchScriptError)));
+        Ok(first_row.get(0))
```

Comment:
Nit: You use `get_checked` in other places. Is there any good reason to not use it here?